src/components/Asset/AssetActions/Download.tsx

```diff
@@ -80,6 +103,30 @@ export default function Download({
     isOwned
   ])
 
+  useEffect(() => {
+    if (!accountId || asset.nft.owner !== accountId) return
+    const queryContext = getQueryContext(Number(asset.chainId))
+
+    async function getBaseTokenBalance() {
+      let baseTokenSum = 0
+      const variables = {
+        user: accountId.toLowerCase()
+      }
+      const result: OperationResult<FixedRateExchanges> = await fetchData(
+        FixedRateExchangesQuery,
+        variables,
+        queryContext
+      )
+      for (let i = 0; i < result.data.fixedRateExchanges.length; i++) {
```

Comment:
this is not right, you are collecting tokens from just one fixed rate exchange, so why are you adding from all fixed rate exchanges?

Reply:
my bad, I've made a confusion. corrected it by querying by user and exchangeId